src/plots/plots.js

```diff
@@ -691,6 +693,39 @@ plots.linkSubplots = function(newFullData, newFullLayout, oldFullData, oldFullLa
             }
         }
     }
+
+    // while we're at it, link overlaying axes to their main axes and
+    // anchored axes to the axes they're anchored to
+    var axList = Plotly.Axes.list(mockGd, null, true);
+    for(i = 0; i < axList.length; i++) {
```

Comment:
Much better. I'd glad `linkSubplots` is getting more attention. Updating all those refs early during `supplyDefaults` sounds like the way to go.

One small non-:no_entry: comment, that we could easily address later: `Axes.list` is now called twice during `Plots.supplyDefaults` here and after `doAutoMargin` [here](https://github.com/plotly/plotly.js/blob/739c9988cd1f8c44126d627bbb25b592f05ff159/src/plots/plots.js#L500). Maybe we could call it once and stash the result in `fullLayout` instead?